client/src/main.rs

```diff
@@ -229,86 +241,35 @@ fn main() {
         .values_of("lua lib search")
         .unwrap_or_default()
         .collect::<Vec<_>>();
-    lua::init_awesome_libraries(&lib_paths);
-    let (display, event_queue, _globals) = init_wayland();
+
+    let (display, mut event_queue) = wayland_obj::init_wayland().unwrap_or_else(|_| std::process::exit(1));
     let (session_fd, system_fd) = dbus::connect().expect("Could not set up dbus connection");
-    init_glib(display, event_queue, session_fd, system_fd);
     let config = matches.value_of("config");
     lua::run_awesome(&lib_paths, config);
-}
 
-fn init_wayland() -> (Display, EventQueue, GlobalManager) {
-    let (display, mut event_queue) = Display::connect_to_env().unwrap_or_else(|err| {
-        match err {
-            ConnectError::NoWaylandLib => {
-                error!("Could not find Wayland library, is it installed and in PATH?")
-            },
-            ConnectError::NoCompositorListening => {
-                error!("Could not connect to Wayland server. Is it running?");
-                error!(
-                    "WAYLAND_DISPLAY={}",
-                    env::var("WAYLAND_DISPLAY").unwrap_or_else(|_| "".into())
-                );
-            },
-            ConnectError::InvalidName => error!("Invalid socket name provided in WAYLAND_SOCKET"),
-            ConnectError::XdgRuntimeDirNotSet => error!("XDG_RUNTIME_DIR must be set"),
-            ConnectError::InvalidFd => error!("Invalid socket provided in WAYLAND_SOCKET")
-        }
-        exit(1);
+    let (display, _globals) = run_wayland(display, &mut event_queue);
+
+    init_glib(display, event_queue, session_fd, system_fd);
+
+    // TODO(ried): hold off refresh until at least one screen is available
+    // (awesome does not like running headless)
```

Comment:
If you could detail what exactly is necessary to make awesome run headless a patch can be constructed that can be sent to them to fix this issue.

Making it in the AwesomeWM bug tracker probably makes the most sense, as then we can easily get feedback before too much work is attempted (e.g. in case they are adverse to such a patch, I don't know why they would, or if they have some insight)